lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -884,9 +929,13 @@ def add_mastercard_network_tokenization_ucaf_data(xml, payment_method, options)
         end
       end
 
-      def add_payment_network_token(xml)
+      def add_payment_network_token(xml, payment_method, options)
+        return unless network_tokenization?(payment_method)
+
+        transaction_type = payment_method.source == :network_token ? '3' : '1'
         xml.tag! 'paymentNetworkToken' do
-          xml.tag!('transactionType', '1')
+          xml.tag!('requestorID', options[:trid]) if transaction_type == '3' && options[:trid]
```

Comment:
Is the `transaction_type == '3'` guard in case an AP/GP has a trid?

Reply:
Yes, just to be extra cautious but I guess they wouldn't have a trid to begin with